platform/linuxbsd/detect.py

```diff
@@ -217,60 +233,64 @@ def configure(env: "Environment"):
         sys.exit(255)
 
     if not env["builtin_freetype"]:
-        env.ParseConfig("pkg-config freetype2 --cflags --libs")
+        add_external_libs(["freetype2"])
```

Comment:
It's a nice cleanup, but won't change the behavior. The `FIXME:` comment was likely referring to adding check if specific libraries are used (modules or features enabled) and displaying errors if required external lib is not found.

Reply:
I'm certainly hoping that it won't change the behavior! In fact, I simply wrapped up the check that had already previously been put in for a few of the later pkg-config lines (i.e. `if os.system("pkg-config --exists fontconfig") == 0`) into a function, so that all of the pkg-config lines pass through that check before attempting to use `env.ConfigParse("pkg-config ...`

Reply:
> The `FIXME:` comment was likely referring to adding check if specific libraries are used (modules or features enabled) and displaying errors if required external lib is not found.

I think that some of the pkg-config lines lower down in the file do either print a warning or an error when not found, e.g.

https://github.com/godotengine/godot/blob/c21c2706ad33c59295f6a3ac1e128db6fa7cce69/platform/linuxbsd/detect.py#L321-L326

and

https://github.com/godotengine/godot/blob/c21c2706ad33c59295f6a3ac1e128db6fa7cce69/platform/linuxbsd/detect.py#L396-L399

The `add_external_libs` function I created could probably be expanded upon to either generate a warning or error when the external lib(s) are not found, but I think this PR can at least act as a good starting point.